src/app/modules/payments/stripe.controller.ts

```diff
@@ -388,3 +423,98 @@ export const handleConnectOauthCallback = [
+    message: 'getPaymentInfo endpoint called',
+    meta: {
+      action: 'getPaymentInfo',
+      paymentId,
+    },
+  })
+
+  return PaymentService.findPaymentById(paymentId)
+    .andThen((payment) => {
+      return PendingSubmissionModel.findPendingSubmissionById(
+        payment.pendingSubmissionId,
+      )
+        .andThen((submission) => FormService.retrieveFormById(submission.form))
+        .andThen((form) => {
+          // Payment forms are encrypted
+          if (!isFormEncryptMode(form)) {
+            logger.error({
+              message:
+                'Requested for payment information in possibly non-payment form. Expected payment forms to be encrypted forms',
+              meta: {
+                action: 'getPaymentInfo',
+                paymentId,
+                formResponseMode: form.responseMode,
+              },
+            })
+            return errAsync(
+              new ResponseModeError(
+                FormResponseMode.Encrypt,
+                form.responseMode,
+              ),
+            )
+          }
+          const stripeAccount = form.payments_channel?.target_account_id
+          // Early termination to prevent consumption of QPS limit to stripe
+          if (!stripeAccount) {
+            logger.error({
+              message: 'Missing payments_channel on this form',
+              meta: {
+                action: 'getPaymentInfo',
+                paymentId,
+              },
+            })
+            return errAsync(new PaymentAccountInformationError())
+          }
+
+          const paymentIntentId = payment.paymentIntentId
+          return ResultAsync.fromPromise(
+            stripe.paymentIntents.retrieve(paymentIntentId, {
+              stripeAccount,
+            }),
+            (error) => {
+              logger.error({
+                message: 'stripe.paymentIntents.retrieve called',
+                meta: {
+                  action: 'getPaymentInfo',
+                  paymentId,
+                  paymentIntentId,
+                  error,
+                },
+              })
+              return new StripeFetchError(String(error))
+            },
+          ).map((paymentIntent) => {
+            return res.status(StatusCodes.OK).json({
+              client_secret: paymentIntent.client_secret || '',
+              publishableKey: form.payments_channel?.publishable_key ?? '',
+              payment_intent_id: payment.paymentIntentId,
+            })
+          })
+        })
+    })
+    .mapErr((error) => {
+      switch (error.constructor) {
+        case ResponseModeError:
+          res.status(StatusCodes.UNPROCESSABLE_ENTITY)
+          break
+        case PendingSubmissionNotFoundError:
+          res.status(StatusCodes.NOT_FOUND)
+          break
+        case StripeFetchError: // fall-through
+        case PaymentAccountInformationError: // fall-through
+        default:
+          res.status(StatusCodes.INTERNAL_SERVER_ERROR)
+          break
+      }
```

Comment:
supernit: (forgot if i mentioned already but) would be nice to extract this to a `mapRouteErr` util, can be reused in other controllers too